### PR TITLE
filter division expression with parentheses (fix #4838)

### LINKED ISSUE #4838: Filter not recognized when filtered expression contains a '/' with ()
### Vue.js version
2.1.10

### Reproduction Link
A variation on #4415 
https://jsbin.com/heguyujupu/edit?html,console,output

### Steps to reproduce
```js
{{ (price) / 100 | currency }}
```

### What is Expected?
This divided number gets filtered

### What is actually happening?
Get this error:

```
[Vue warn]: Property or method \"currency\" is not defined on the instance but referenced during render. Make sure to declare reactive data properties in the data option. 
(found in root instance)"
```

### PR DESCRIPTION
**Pull request by rixlabs**
The filter parser wasn't considering "closing" parentheses in the check for regex.

Fix for issue #4838

### CODE REVIEW

**Comment by rixlabs** [comment]
the last commit message is wrong. I mean that I miss a lint check that's why the check failed before

**Comment by posva** [comment]
@rixlabs Don't worry for that. We'll probably squash it just do one, clean, commit 😄 
Thanks for taking the time to look into this!

File: test/unit/features/filter/filter.spec.js [hunk 1]
```diff
@@ -78,6 +78,41 @@ describe('Filters', () => {
...
+      data: { a: 20 },
+      template: `<div>{{ (a*2) / 5 | double }}</div>`,
+      filters: { double: v => v * 2 }
+    }).$mount()
+    expect(vm.$el.textContent).toBe(String(16))
+  })
+
+  it('handle division with dot', () => {
+    const vm = new Vue({
+      template: `<div>{{ 20. / 5 | double }}</div>`,
```
Comment (HerringtonDarkholme) [inline test/unit/features/filter/filter.spec.js]: what about `{{ a++ / 5 }}` ? and `{{ privateVar__  / 5}}`

Comment (rixlabs) [inline test/unit/features/filter/filter.spec.js]: I didn't considered + and - and also _  I have to include those too

Comment (rixlabs) [inline test/unit/features/filter/filter.spec.js]: made a quick test checking every single exception with `/[\w).\]\+\-\_$]/` maybe there is a more generic way to do this

**Comment by rixlabs** [comment]
now I also search for + - _ also sync the repo

**Comment by yyx990803** [comment]
Thanks!

**Pull request merged**